### ISSUE TITLE
False positive warning when poping from a open TypedDict

### ISSUE THREAD

**Issue by Daraan** [microsoft/pyright#9805]
**Describe the bug**
I am using `dict.pop(key, default)` to remove an extra item from a `TypedDict` this results in an unexpected error, that seems wrong: I think `pop` misses an overload for a simple `str` in this case.
Pylance shows all provided keys from the `TypedDict` in the signature.

```python
error: No overloads for "pop" match the provided arguments
error: Argument of type "Literal['poped_key']" cannot be assigned to parameter "k" of type "Literal['some_present_key']" in function "pop"
  "Literal['poped_key']" is not assignable to type "Literal['some_present_key']"
```

**Code or Screenshots**
Code sample in [pyright playground](https://pyright-play.net/?enableExperimentalFeatures=true&code=GYJw9gtgBALgngBwJYDsDmB9ApgDxllAZyTCKiQgTBBigBVEsATAESQGMYAaKAJSwCOAVyQhmPAHJgY-YaPFQAgijg8GCLIoA2SAIaEAUAfZb9hJQAp1zNpx4xpurQF4AYk8JYeJsJ6ZuPLABKAC4DKAioYDAwEKitMF0YIyYsYCg0LBgMRQsgqABaAD4lOIA6CqNdKGcMrJy8qrKqBAsAIkJILABrLDg2yVJggyA)

```python
from typing_extensions import TypedDict, Required, NotRequired, Any, TypeAlias

class A(TypedDict, total=False, closed=False):
    foo: float
    

def get_A() -> A: ...

a = get_A()

a.pop("somekey", None)

```

**VS Code extension or command-line**
pyright 1.1.393

**Comment by erictraut** [comment]
The typing spec doesn't specify whether type checkers should synthesize fallback method overloads for keys that are unspecified in a TypedDict definition. Mypy, which was the original reference implementation for TypedDict, makes the conservative assumption and doesn't synthesize any methods (`get`, `pop`, `__getitem__`, etc.) for items that are not defined in the TypedDict. Pyright is less conservative and synthesizes fallback methods for `get`, `__getitem__`, `__setitem__` and `__delitem__`, but it doesn't currently do the same for `pop`. I think it makes sense to add this for consistency.

This will be addressed in the next release.

**Comment by erictraut** [comment]
This is addressed in pyright 1.1.394.
